searchlib/src/main/java/com/yahoo/searchlib/rankingexpression/evaluation/ExpressionOptimizer.java

```diff
@@ -44,12 +44,16 @@ public Optimizer getOptimizer(Class<?> clazz) {
         return null;
     }
 
-    public OptimizationReport optimize(RankingExpression expression, ContextIndex arrayContext) {
+    public OptimizationReport optimize(RankingExpression expression, ContextIndex contextIndex) {
         OptimizationReport report = new OptimizationReport();
         // Note: Order of optimizations matter
-        gbdtOptimizer.optimize(expression, arrayContext, report);
-        gbdtForestOptimizer.optimize(expression, arrayContext, report);
+        gbdtOptimizer.optimize(expression, contextIndex, report);
+        gbdtForestOptimizer.optimize(expression, contextIndex, report);
         return report;
     }
 
+    public OptimizationReport optimize(RankingExpression expression, AbstractArrayContext arrayContext) {
```

Comment:
perhaps deprecate this method as we want to remove it in near future?

Reply:
Yes, I'll do that